contracts/TapETH.sol

```diff
@@ -0,0 +1,544 @@
+        _emitTransferEvents(_sender, _recipient, _amount, _sharesToTransfer);
+    }
+
+    /**
+     * @notice Sets `_amount` as the allowance of `_spender` over the `_owner` s tokens.
+     *
+     * Emits an `Approval` event.
+     */
+    function _approve(
+        address _owner,
+        address _spender,
+        uint256 _amount
+    ) internal {
+        require(_owner != address(0), "TapETH: APPROVE_FROM_ZERO_ADDR");
+        require(_spender != address(0), "TapETH: APPROVE_TO_ZERO_ADDR");
+
+        allowances[_owner][_spender] = _amount;
+        emit Approval(_owner, _spender, _amount);
+    }
+
+    /**
+     * @dev Updates `owner` s allowance for `spender` based on spent `amount`.
+     *
+     * Does not update the allowance amount in case of infinite allowance.
+     * Revert if not enough allowance is available.
+     *
+     * Might emit an {Approval} event.
+     */
+    function _spendAllowance(
+        address _owner,
+        address _spender,
+        uint256 _amount
+    ) internal {
+        uint256 currentAllowance = allowances[_owner][_spender];
+        if (currentAllowance != INFINITE_ALLOWANCE) {
+            if (currentAllowance < _amount) {
+                revert InsufficientAllowance(currentAllowance, _amount);
+            }
+
+            _approve(_owner, _spender, currentAllowance - _amount);
+        }
+    }
+
+    /**
+     * @return the total amount of shares in existence.
+     */
+    function _getTotalShares() internal view returns (uint256) {
+        return totalShares;
+    }
+
+    /**
+     * @return the amount of shares owned by `_account`.
+     */
+    function _sharesOf(address _account) internal view returns (uint256) {
+        return shares[_account];
+    }
+
+    /**
+     * @notice Moves `_sharesAmount` shares from `_sender` to `_recipient`.
+     */
+    function _transferShares(
+        address _sender,
+        address _recipient,
+        uint256 _sharesAmount
+    ) internal {
+        require(_sender != address(0), "TapETH: zero address");
+        require(_recipient != address(0), "TapETH: zero address");
+        require(
+            _recipient != address(this),
+            "TapETH: TRANSFER_TO_tapETH_CONTRACT"
+        );
+
+        uint256 currentSenderShares = shares[_sender];
+
+        if (_sharesAmount > currentSenderShares) {
+            revert InsufficientBalance(currentSenderShares, _sharesAmount);
+        }
+
+        shares[_sender] -= _sharesAmount;
+        shares[_recipient] += _sharesAmount;
+    }
+
+    /**
+     * @notice Creates `_sharesAmount` shares and assigns them to `_recipient`, increasing the total amount of shares.
+     */
+    function _mintShares(
+        address _recipient,
+        uint256 _sharesAmount
+    ) internal returns (uint256 newTotalShares) {
+        require(_recipient != address(0), "TapETH: MINT_TO_ZERO_ADDR");
+        uint256 _tokenAmount = getPooledEthByShares(_sharesAmount);
```

Comment:
mint wouldn't work for an empty pool and there isn't a way out of this. Let's say we mint 1 wei of shares, the `_tokenAmount ` will be `0`, and this means `_totalSupply ` isn't properly maintained.

Reply:
I identified the issue, I m working to fix it.